src/cal/mod.rs

```diff
@@ -0,0 +1,176 @@
+use std::collections::BTreeSet;
+use std::ops::Range;
+
+use chrono::DateTime;
+use chrono::Duration;
+use chrono::Utc;
+
+use std::cmp::Ordering;
+
+pub struct Cal {
+    events: BTreeSet<CmpEvent>,
+}
+
+#[allow(dead_code)]
+impl Cal {
+    pub fn new() -> Cal {
+        Cal {
+            events: BTreeSet::new(),
+        }
+    }
+
+    pub fn events_in(&self, range: Range<DateTime<Utc>>) -> impl Iterator<Item = Event> + '_ {
+        let event_range = Range {
+            start: CmpEvent::from_date(range.start),
+            end: CmpEvent::from_date(range.end),
+        };
+        self.events.range(event_range).map(|x| x.event.clone())
```

Comment:
Is the `clone()` necessary? I think this is more flexible if you return an iterator to `Event` references. Then the caller can clone them if they need to, and can avoid the cloning overhead if not.

Reply:
also: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.cloned